packages/wrangler/src/queues/client.ts

```diff
@@ -46,103 +48,148 @@ export type Consumer = ScriptReference & {
 	type: string;
 };
 
-export interface QueueResponse {
-	queue_id: string;
+export interface TypedConsumerResponse extends Consumer {
 	queue_name: string;
 	created_on: string;
-	modified_on: string;
-	producers: ScriptReference[];
-	producers_total_count: number;
-	consumers: Consumer[];
-	consumers_total_count: number;
-	settings?: QueueSettings;
+}
+
+export interface PostTypedConsumerBody {
+	type: string;
+	script_name?: string;
+	environment_name?: string;
+	settings: ConsumerSettings;
+	dead_letter_queue?: string;
+}
+
+export interface ConsumerSettings {
+	batch_size?: number;
+	max_retries?: number;
+	max_wait_time_ms?: number;
+	max_concurrency?: number | null;
+	visibility_timeout_ms?: number;
+	retry_delay?: number;
+}
+
+export async function createQueue(
+	config: Config,
+	body: PostQueueBody
+): Promise<QueueResponse> {
+	const accountId = await requireAuth(config);
+	return fetchResult(`/accounts/${accountId}/queues`, {
+		method: "POST",
+		body: JSON.stringify(body),
+	});
 }
 
 export async function deleteQueue(
 	config: Config,
 	queueName: string
+): Promise<void> {
+	const queue = await getQueue(config, queueName);
+	return deleteQueueById(config, queue.queue_id);
+}
+
+export async function deleteQueueById(
+	config: Config,
+	queueId: string
 ): Promise<void> {
 	const accountId = await requireAuth(config);
-	return await fetchResult(
-		`/accounts/${accountId}/workers/queues/${queueName}`,
-		{
-			method: "DELETE",
-		}
-	);
+	return fetchResult(`/accounts/${accountId}/queues/${queueId}`, {
+		method: "DELETE",
+	});
 }
 
 // TODO(soon) show detailed queue response
 export async function listQueues(
 	config: Config,
-	page?: number
+	page?: number,
+	queueName?: string
 ): Promise<QueueResponse[]> {
 	page = page ?? 1;
 	const accountId = await requireAuth(config);
-	return await fetchResult(
-		`/accounts/${accountId}/workers/queues`,
-		{},
-		new URLSearchParams({ page: page.toString() })
-	);
+	const params = new URLSearchParams({ page: page.toString() });
+
+	if (queueName) {
+		// TODO final parameter name
+		params.append("queueName", queueName);
+	}
+
+	return fetchResult(`/accounts/${accountId}/queues`, {}, params);
```

Comment:
Should we factor out the hard-coded URLs into something like 

```ts
function queuesURL(accountId: string, queueId?: string): string {
  let url = "/accounts/${accountId}/queues"
  if (queueId) {
    url = `${url}/${queueId}`
  }
  return url
} 
```

?

Reply:
Good call.

Reply:
Added a small URL builder.